### PR TITLE
Fix: Nav active fix

### LINKED ISSUE #917: Pagination numbering issue and inactive side navigation links on mobile
### Pagination numbering

The pagination on the project list page should display numbers as 1, 2, 3...10 instead of 1, 2, 3, 4, 5, 6, 7, 8, 9, 10. This will provide better clarity on the total number of pages available.

image

### Inactive side navigation links on mobile

On the mobile version of our website, the side navigation links are not active when clicked. This prevents users from accessing the corresponding sections of the page.
image

### Steps color in dark mode on the documentation page

Currently, in dark mode on the documentation page, the color of the steps is not consistent with the overall theme. To improve the visual experience, the steps should have a light purple color when viewed in dark mode.

image

### PR DESCRIPTION
**Pull request by manaver**
This pull request fixes #917 adds the Nav Active feature for mobile devices to enhance the user experience. The Nav Active feature ensures that the active menu item is visually highlighted when accessed on mobile devices, providing better navigation and usability.

### CODE REVIEW

File: src/components/SideMenu.jsx [hunk 1]
```diff
@@ -6,6 +6,40 @@ import { ThemeContext } from '../context/Theme';
...
+      path: '/projects',
+    },
+    {
+      name: 'Documentation',
+      path: '/docs',
+    },
+  ];
+
+  const sideLinkEls = sideLinks.map((sideLink) => (
+    <li>
```
Comment (priyankarpal) [inline src/components/SideMenu.jsx:26]: remove `li`
``` suggestion

```

File: src/components/SideMenu.jsx [hunk 2]
```diff
@@ -6,6 +6,40 @@ import { ThemeContext } from '../context/Theme';
...
+      <NavLink
```
Comment (priyankarpal) [inline src/components/SideMenu.jsx:27]: addd `key={i}`

```suggestion
      <NavLink 
      key={i}
```

File: src/components/SideMenu.jsx [hunk 3]
```diff
@@ -6,6 +6,40 @@ import { ThemeContext } from '../context/Theme';
...
+      name: 'Projects',
+      path: '/projects',
+    },
+    {
+      name: 'Documentation',
+      path: '/docs',
+    },
+  ];
+
+  const sideLinkEls = sideLinks.map((sideLink) => (
```
Comment (priyankarpal) [inline src/components/SideMenu.jsx:25]:
```suggestion
  const sideLinkEls = sideLinks.map((sideLink,i) => (
```

File: src/components/SideMenu.jsx [hunk 4]
```diff
@@ -6,6 +6,40 @@ import { ThemeContext } from '../context/Theme';
...
+    <li>
+      <NavLink
+        to={sideLink.path}
+        className={({ isActive }) =>
+          `inline-block py-2 px-3 m-2 text-center font-bold  rounded-md ${
```
Comment (priyankarpal) [inline src/components/SideMenu.jsx:30]:
```suggestion
          `flex py-2 px-3 m-2 text-center font-bold  rounded-md ${
```

**Review by priyankarpal — CHANGES_REQUESTED** [review]
also, change the PR title to `Fix: Nav active fix`

File: src/components/SideMenu.jsx [hunk 5]
```diff
@@ -6,6 +6,40 @@ import { ThemeContext } from '../context/Theme';
...
+            theme.mode === 'light'
+              ? `hover:text-white hover:bg-black transition-all duration-200 ${isActive && 'text-white bg-black'}`
+              : `hover:text-black hover:bg-white transition-all duration-200 ${isActive && 'text-black bg-white'}`
+          }`
+        }
+        onClick={() => props.handleDrawerToggle()}
+      >
+        {sideLink.name}
+      </NavLink>
+    </li>
```
Comment (priyankarpal) [inline src/components/SideMenu.jsx:40]: also, remove `li` from here.
```suggestion
    
```

**Review by priyankarpal — CHANGES_REQUESTED** [review]

File: src/components/SideMenu.jsx (hunk 5 above)
Comment (manaver) [reply src/components/SideMenu.jsx:40]: But they are list elements?

**Comment by manaver** [comment]
hi, @priyankarpal  i made these changes as requested🙂

**Pull request merged**